File: 2 D Array/Matrix Product/matrixProduct.c
// Program to compute the product of two matrices

#include<stdio.h>
#define maxSize 100
int main()
{
    int matrixOne[maxSize][maxSize], size;
    int matrixTwo[maxSize][maxSize];
    int product[maxSize][maxSize];

    // Scanning the size of the matrix 
    printf("Enter the size of matrix: ");
    scanf("%d",&size);

    // Scanning the matrix I
    printf("Enter the elements of matrix I:\n");
    for(int i=0; i<size; i++)
    {
        for(int j=0; j<size; j++)
        {
            printf("Element [%d][%d]: ",i,j);
            scanf("%d",&matrixOne[i][j]);
        }
    }

    // Scanning the matrix II
    printf("Enter the elements of matrix II:\n");
    for(int i=0; i<size; i++)
    {
        for(int j=0; j<size; j++)
        {
            printf("Element [%d][%d]: ",i,j);
            scanf("%d",&matrixTwo[i][j]);
        }
    }

    // Printing matrix I
    printf("Matrix I: \n");
    for(int i=0; i<size; i++)
    {
        for(int j=0; j<size; j++)
        {
            printf("%d ",matrixOne[i][j]);
        }
        printf("\n");
    } 

    // Printing matrix II
    printf("Matrix II: \n");
    for(int i=0; i<size; i++)
    {
        for(int j=0; j<size; j++)
        {
            printf("%d ",matrixTwo[i][j]);
        }
        printf("\n");
    } 

    // Computing the product
    for(int i=0; i<size; i++)
    {
        for(int j=0; j<size; j++)
        {
            product[i][j] = 0;
            for(int k=0; k<size; k++)
            {
                product[i][j] += matrixOne[i][k] * matrixTwo[k][j];
            }
        }
    }

    // Printing the product of the matrices
    printf("PRODUCT OF MATRICES: \n");
    for(int i=0; i<size; i++)
    {
        for(int j=0; j<size; j++)
        {
            printf("%d ",product[i][j]);
        }
        printf("\n");
    }

    return 0;
}

File: 2 D Array/Matrix Product/matrixProduct.java
// Program to compute the product of two matrices

import java.util.*;

public class matrixProduct 
{
    public static void main(String[] args)
    {
        try(Scanner sc = new Scanner(System.in))
        {
            // Scanning the size of the matrix
            System.out.println("Enter the size of the matrix: ");
            int size = sc.nextInt();

            int matrixOne[][] = new int[size][size];
            // Scanning the elements of matrix I
            System.out.println("Enter the elements of matrix I: ");
            for(int i=0; i<size; i++)
            {
                for(int j=0; j<size; j++)
                {
                    System.out.print("Element ["+ i +"]["+ j +"]: ");
                    matrixOne[i][j] = sc.nextInt();
                }
            }

            int matrixTwo[][] = new int[size][size];
            // Scanning the elements of matrix II
            System.out.println("Enter teh elements of matrix II ");
            for(int i=0; i<size; i++)
            {
                for(int j=0; j<size; j++)
                {
                    System.out.print("Element ["+ i +"]["+ j +"]: ");
                    matrixTwo[i][j] = sc.nextInt();
                }
            }

            // Printing matrix I
            System.out.println("Matrix I: ");
            for(int i=0; i<size; i++)
            {
                for(int j=0; j<size; j++)
                {
                    System.out.print(matrixOne[i][j] + " ");
                }
                System.out.println();
            }

            // Printing matrix II
            System.out.println("Matrix II: ");
            for(int i=0; i<size; i++)
            {
                for(int j=0; j<size; j++)
                {
                    System.out.print(matrixTwo[i][j] + " ");
                }
                System.out.println();
            }

            int product[][] = new int[size][size];
            // Computing the product of matrices
            for(int i=0; i<size; i++)
            {
                for(int j=0; j<size; j++)
                {
                    product[i][j] = 0;
                    for(int k=0; k<size; k++)
                    {
                        product[i][j] += matrixOne[i][k] * matrixTwo[k][j];
                    }
                }
            }

            // Printing the product of matrices
            System.out.println("Product of matrices: ");
            for(int i=0; i<size; i++)
            {
                for(int j=0; j<size; j++)
                {
                    System.out.print(product[i][j] + " ");
                }

                System.out.println();
            }
        }
    }    
}
